### PR TITLE
506 : Implement Product Catalogue API

### PR DESCRIPTION
**Pull request by rehammuzzamil**

### CODE REVIEW

**Review by dubdabasoduba — DISMISSED** [review]

**Review by dubdabasoduba — APPROVED** [review]

**Review by dubdabasoduba — APPROVED** [review]

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java [hunk 1]
```diff
@@ -0,0 +1,157 @@
...
+			logger.error(
+					String.format("Exception occurred while persisting image of Product Catalogue" + e.getMessage() + e));
+			return new ResponseEntity<String>("Exception occurred while persisting image of Product Catalogue",
+					HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		catch (IllegalArgumentException e) {
+			logger.error(String.format("Exception occurred while adding product catalogue: %s,%s", e.getMessage(), e));
+			return new ResponseEntity<String>("The request contain illegal argument : " + e.getMessage(),
+					HttpStatus.BAD_REQUEST);
+		}
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java:97]: This should be handled by Global exception handler

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java [hunk 2]
```diff
@@ -0,0 +1,157 @@
...
+	@Autowired
+	private ProductCatalogueService productCatalogueService;
+
+	@Autowired
+	private MultimediaService multimediaService;
+
+	private static Logger logger = LoggerFactory.getLogger(ProductCatalogueResource.class.toString());
+
+	@GetMapping(produces = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<List<ProductCatalogue>> getAll(
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java:45]: Can we just return `List<ProductCatalogue>` here

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java [hunk 3]
```diff
@@ -0,0 +1,157 @@
...
+			logger.error(
+					String.format("Exception occurred while persisting image of Product Catalogue" + e.getMessage() + e));
+			return new ResponseEntity<String>("Exception occurred while persisting image of Product Catalogue",
+					HttpStatus.INTERNAL_SERVER_ERROR);
+		}
+		catch (IllegalArgumentException e) {
+			logger.error(String.format("Exception occurred while updating product catalogue: %s", e.getMessage()));
+			return new ResponseEntity<String>("The request contain illegal argument : " + e.getMessage(),
+					HttpStatus.BAD_REQUEST);
+		}
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java:134]: This should be handled by Global Exception handler

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+	@DeleteMapping(value = "/{id}", produces = { MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<String> delete(@PathVariable("id") Long uniqueId) {
+		try {
+			productCatalogueService.deleteProductCatalogueById(uniqueId);
+			return new ResponseEntity<>(HttpStatus.NO_CONTENT);
+		}
+		catch (IllegalArgumentException e) {
+			logger.error(e.getMessage(), e);
+			return new ResponseEntity<>(e.getMessage(), HttpStatus.BAD_REQUEST);
+		}
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java:146]: This should be handled by Global Exception handler

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java [hunk 5]
```diff
@@ -0,0 +1,157 @@
...
+
+	}
+
+	@GetMapping(value = "/{id}", produces = {
+			MediaType.APPLICATION_JSON_VALUE })
+	public ResponseEntity<ProductCatalogue> getByUniqueId(@PathVariable("id") Long uniqueId) {
```
Comment (githengi) [inline src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java:152]: Return `ProductCatalogue`

File: src/main/java/org/opensrp/web/config/multipartresolver/PostAndPutCommonsMultipartResolver.java [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+	private static final String POST_METHOD = "POST";
+	private static final String PUT_METHOD = "PUT";
+
+	@Override
+	public boolean isMultipart(HttpServletRequest request) {
+
+		boolean isMultipartRequest = false;
+
+		if (request != null && POST_METHOD.equalsIgnoreCase(request.getMethod()) || PUT_METHOD
+				.equalsIgnoreCase(request.getMethod())) {
```
Comment (githengi) [inline src/main/java/org/opensrp/web/config/multipartresolver/PostAndPutCommonsMultipartResolver.java:20]:
```suggestion
		if (request != null && (POST_METHOD.equalsIgnoreCase(request.getMethod()) || PUT_METHOD
				.equalsIgnoreCase(request.getMethod()))) {
```

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java (hunk 1 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java]: Removed

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java (hunk 2 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java]: Done

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java (hunk 3 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java]: Removed this catch block.

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java (hunk 4 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java]: Removed.

File: src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java (hunk 5 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/rest/ProductCatalogueResource.java]: Done.

File: src/main/java/org/opensrp/web/config/multipartresolver/PostAndPutCommonsMultipartResolver.java (hunk 6 above)
Comment (rehammuzzamil) [reply src/main/java/org/opensrp/web/config/multipartresolver/PostAndPutCommonsMultipartResolver.java]: Updated.

**Review by githengi — DISMISSED** [review]

**Review by githengi — APPROVED** [review]

**Pull request merged**